### PR TITLE
SPARK-532493 webrtc core should expose separate events for ice and connection state

### PR DESCRIPTION
**Pull request by k-wasniowski**
This PR divide evaluating of overall connection state to two separated events.
We have decided that it's useful to have access to both events IceConnectionState and ConnectionState separately.
It's required for JS-SDK to have access directly to the reason of media join failure.

### CODE REVIEW

File: src/connection-state-handler.ts [hunk 1]
```diff
@@ -113,6 +122,32 @@ export class ConnectionStateHandler extends EventEmitter<ConnectionStateEventHan
...
+   * Gets current ice connection state.
+   *
+   * @returns Current ice connection state.
+   */
+  public getIceConnectionState(): IceConnectionState {
+    const { iceState } = this.getCurrentStatesCallback();
+
+    const state = iceState[0].toUpperCase() + iceState.slice(1);
+
+    return state as IceConnectionState;
```
Comment (marcin-bazyl) [inline src/connection-state-handler.ts:142]: I'm not sure why we need to define our own `IceConnectionState` enum. Wouldn't it be better to have `RTCIceConnectionState` as the return type for `getIceConnectionState()` and `RTCPeerConnectionState` as return type for `getConnectionState()`? Then we wouldn't have to be doing these tricks with converting first letter to uppercase etc. which I'm not a huge fan of....

File: src/connection-state-handler.ts [hunk 2]
```diff
@@ -113,6 +122,32 @@ export class ConnectionStateHandler extends EventEmitter<ConnectionStateEventHan
...
+  }
+
+  /**
+   * Gets current overall connection state.
+   *
+   * @returns Current overall connection state.
+   */
+  public getOverallConnectionState(): OverallConnectionState {
```
Comment (brycetham) [inline src/connection-state-handler.ts:150]: Wondering if it makes sense to keep calling this the `connectionState` and have the 2 sub-states be called `peerConnectionState` and `iceConnectionState`?

**Review by brycetham — COMMENTED** [review]
Agreed with @marcin-bazyl's comments. Also left a comment about naming.

File: src/connection-state-handler.ts (hunk 2 above)
Comment (k-wasniowski) [reply src/connection-state-handler.ts]: Changed it as suggested. I like it, however had to change event name to keep it consistent. What do you think about it?

File: src/connection-state-handler.ts (hunk 1 above)
Comment (k-wasniowski) [reply src/connection-state-handler.ts]: Done

**Review by marcin-bazyl — DISMISSED** [review]

File: src/connection-state-handler.spec.ts [hunk 3]
```diff
@@ -17,39 +17,53 @@ describe('ConnectionStateHandler', () => {
...
+
+    expect(connStateHandler.getIceConnectionState()).toBe('new');
+  });
+
   it('reads initial connection state', () => {
     expect.assertions(1);
     const connStateHandler = new ConnectionStateHandler(fakeCallback);
 
-    expect(connStateHandler.getConnectionState()).toStrictEqual(ConnectionState.New);
+    expect(connStateHandler.getConnectionState()).toBe('New');
```
Comment (brycetham) [inline src/connection-state-handler.spec.ts:38]: Should this still be `ConnectionState.New`?

**Review by brycetham — APPROVED** [review]
Looks good, left a minor comment. As I discussed with you offline, since this is technically a breaking change, I would check with the SDK team to make sure they are ok with it, as this might affect some SDK customers. Likely they will need to document this somewhere, so I might wait until you get the ok before merging.

**Review by szotrabh — APPROVED** [review]

**Pull request merged**